File: frontend/src/app/types/window.ts
export type WindowStatus = 'candidate' | 'submitted' | 'locked' | 'allocated' | 'cancelled';

export const WINDOW_STATUSES: WindowStatus[] = ['candidate', 'submitted', 'locked', 'allocated', 'cancelled'];

export interface ContactWindow {
  id: number;
  station_id: number;
  station_code: string;
  satellite_id: number;
  satellite_code: string;
  start_at: string;
  end_at: string;
  duration_sec: number;
  band: string;
  elevation_peak_deg: number;
  window_status: WindowStatus;
  priority: number;
  locked: boolean;
  source_version: string;
  version: number;
  updated_at: string;
}

export interface ContactWindowInput {
  station_id: number;
  satellite_id: number;
  start_at: string;
  end_at: string;
  band: string;
  elevation_peak_deg: number;
  priority: number;
  source_version: string;
}

File: backend/internal/constants/window.go
package constants

const (
	WindowStatusCandidate = "candidate"
	WindowStatusSubmitted = "submitted"
	WindowStatusLocked    = "locked"
	WindowStatusAllocated = "allocated"
	WindowStatusCancelled = "cancelled"
)

var WindowStatuses = []string{
	WindowStatusCandidate,
	WindowStatusSubmitted,
	WindowStatusLocked,
	WindowStatusAllocated,
	WindowStatusCancelled,
}

func IsWindowStatus(value string) bool {
	for _, candidate := range WindowStatuses {
		if value == candidate {
			return true
		}
	}
	return false
}

func CanTransitionWindow(from, to string) bool {
	if from == WindowStatusCandidate {
		if to == WindowStatusLocked {
			return true
		}
		return false
	}
	if from == WindowStatusSubmitted {
		return false
	}
	if from == WindowStatusLocked {
		if to == WindowStatusAllocated {
			return true
		}
		if to == WindowStatusCancelled {
			return true
		}
		return false
	}
	if from == WindowStatusAllocated {
		if to == WindowStatusCancelled {
			return true
		}
		return false
	}
	return false
}
